Lib/argparse.py

```diff
@@ -2555,9 +2559,23 @@ def _get_value(self, action, arg_string):
     def _check_value(self, action, value):
         # converted value must be one of the choices (if specified)
         if action.choices is not None and value not in action.choices:
-            args = {'value': value,
-                    'choices': ', '.join(map(repr, action.choices))}
+            args = {
+                'value': value,
+                'choices': ', '.join(map(repr, action.choices)),
+            }
             msg = _('invalid choice: %(value)r (choose from %(choices)s)')
+
+            all_strings = all(isinstance(choice, str) for choice in action.choices)
+
+            if self.suggest_on_error and isinstance(value, str) and all_strings:  
```

Comment:
Thank you.

If `suggest_on_error` is false (by default), you do not even need to check types. So it is better to evaluate `all(isinstance(choice, str) for choice in action.choices)` after testing `self.suggest_on_error and isinstance(value, str)`.

Reply:
Oh good catch - thanks!